Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        // Conduits = 9, Pumps = 7, Weirs = 8
+        if (split_str.size() < 7)
+        {
+            ERR ("Conduit format not recognised.");
+            return false;
+        }
+
+        std::string const inlet (split_str[1]);
+        auto const i_it = name_id_map.find(inlet);
+        if (i_it == name_id_map.end())
+        {
+            ERR ("SwmmInterface::readLineElements(): Inlet node %s not found coordinates map.", inlet.c_str());
+            return false;
+        }
+
+        std::string const outlet (split_str[2]);
+        auto const o_it = name_id_map.find(outlet);
+        if (o_it == name_id_map.end())
+        {
+            ERR ("SwmmInterface::readLineElements(): Outlet node %s not found coordinates map.", outlet.c_str());
+            return false;
+        }
+
+        std::array<MeshLib::Node*, 2> const line_nodes = { nodes[i_it->second], nodes[o_it->second] };
+        elements.push_back(new MeshLib::Line(line_nodes));
+        _id_linkname_map.push_back(split_str[0]);
+        getline(in, line);
+    }
+    return true;
+}
+
+bool SwmmInterface::readSubcatchments(std::ifstream &in, std::map< std::string, std::size_t> const& name_id_map)
+{
+    std::string line;
+    getline(in, line);
+    while (!isSectionFinished(line))
+    {
+        if (isCommentLine(line))
+        {
+            getline(in, line);
+            continue;
+        }
+
+        std::vector<std::string> const split_str (BaseLib::splitString(line));
+        if (split_str.size() < 8)
+        {
+            ERR ("Subcatchment format not recognised.");
+            return false;
+        }
+
+        Subcatchment sc;
+        sc.name = split_str[0];
+        sc.rain_gauge = std::numeric_limits<std::size_t>::max();
+        std::size_t const n_gauges (_rain_gauges.size());
+        for (std::size_t i=0; i<n_gauges; ++i)
+            if (_rain_gauges[i].first.getName() == split_str[1])
+                sc.rain_gauge = i;
+
+        if (sc.rain_gauge == std::numeric_limits<std::size_t>::max())
+        {
+            ERR ("Rain gauge for subcatchment \"%s\" not found.", split_str[0].c_str());
+            return false;
+        }
+
+        sc.outlet =  std::numeric_limits<std::size_t>::max();
+        auto const it = name_id_map.find(split_str[2]);
+        if (it == name_id_map.end())
+        {
+            ERR ("Outlet node for subcatchment \"%s\" not found.", split_str[0].c_str());
+            return false;
+        }
+        sc.outlet = it->second;
+        sc.area = BaseLib::str2number<double>(split_str[3]);
+        _subcatchments.push_back(sc);
+        getline(in, line);
+    }
+
+    return true;
+}
+
+bool SwmmInterface::convertSwmmInputToLineMesh()
```

Comment:
Maybe better call the method readSwmmInput? "Convert" sounds like read, convert, write, forget.